lib/rich-text-codemirror.js

```diff
@@ -943,8 +965,39 @@ firepad.RichTextCodeMirror = (function () {
         this.currentAttributes_[attr] = attributes[attr];
       }
     }
+
+    this.updateToolbarButnsState_();
   };
 
+  RichTextCodeMirror.prototype.updateToolbarButnsState_ = function() {
+    var toolbar=this.codeMirror.firepad.toolbar;
+    var attr=this.currentAttributes_;
+
+    if (!toolbar.boldBtnElem) { // cache button elements
+      toolbar.boldBtnElem = toolbar.element_.getElementsByClassName('firepad-tb-bold')[0].parentElement;
+      toolbar.italicBtnElem = toolbar.element_.getElementsByClassName('firepad-tb-italic')[0].parentElement;
+      toolbar.underlineBtnElem = toolbar.element_.getElementsByClassName('firepad-tb-underline')[0].parentElement;
+      toolbar.strikeBtnElem = toolbar.element_.getElementsByClassName('firepad-tb-strikethrough')[0].parentElement;
+    }
+
+    if (toolbar.updateToolbarButnsStateTimeout) clearTimeout(toolbar.updateToolbarButnsStateTimeout);
+    toolbar.updateToolbarButnsStateTimeout = setTimeout(function() {
+      toggleClass(toolbar.boldBtnElem, 'firepad-btn-highlight', attr.b === true);
+      toggleClass(toolbar.italicBtnElem, 'firepad-btn-highlight', attr.i === true);
+      toggleClass(toolbar.underlineBtnElem, 'firepad-btn-highlight', attr.u === true);
+      toggleClass(toolbar.strikeBtnElem, 'firepad-btn-highlight', attr.s === true);
+    }, 200);
+  }
+
+  // If newState is provided add/remove theClass accordingly, otherwise toggle theClass
+  function toggleClass(elem, theClass, newState) {
+    let matchRegExp = new RegExp('(?:^|\\s)' + theClass + '(?!\\S)', 'g');
```

Comment:
Unless we start building with babelify, you can't use `let` here, it breaks the build.